src/sparsezoo/refactor/directory.py

```diff
@@ -0,0 +1,194 @@
+        :param archive_directory: the local path to
+            save new tar archive Directory to (default = None)
+        :return: tar archive Directory
+        """
+        if directory._is_tar():
+            raise ValueError(
+                "Attempting to create a tar archive "
+                "Directory from a tar archive Directory."
+            )
+        if archive_directory is not None:
+            parent_path = archive_directory
+        else:
+            if directory.path is None:
+                raise ValueError(
+                    "Attempting to zip the folder Directory object files using "
+                    "`path` attribute, but `self.path` is None. "
+                    "Class object requires pointer to parent "
+                    "folder directory to know where to save the tar archive file."
+                )
+            parent_path = pathlib.PurePath(directory.path).parent
+
+        tar_file_name = directory.name + ".tar.gz"
+        tar_file_path = os.path.join(parent_path, tar_file_name)
+        with tarfile.open(tar_file_path, "w") as tar:
+            for file in directory.files:
+                tar.add(file.path)
+        return Directory(name=tar_file_name, path=tar_file_path)
+
+    @classmethod
+    def unzip(
+        cls, tar_directory: "Directory", extract_directory: Optional[str] = None
+    ) -> "Directory":
+        """
+        Factory method that extracts a tar archive `tar_directory` into a new
+        folder Directory.
+        The extracted files would be saved in the parent directory of
+        the `tar_directory`, unless `extract_directory` argument is specified.
+
+        :param tar_directory: the tar archive directory (Directory class object)
+        :param extract_directory: the local path to create
+            folder Directory at (and thus save extracted files at)
+        :return: folder Directory
+        """
+        files = []
+        if extract_directory is None:
+            extract_directory = os.path.dirname(tar_directory.path)
+
+        if not tar_directory._is_tar():
+            raise ValueError(
+                "Attempting to extract tar archive, "
+                "but the Directory object is not tar archive"
+            )
+
+        name = os.path.basename(extract_directory)
+        tar = tarfile.open(tar_directory.path, "r")
+        path = extract_directory
+        for member in tar.getmembers():
+            member.name = os.path.basename(member.name)
+            tar.extract(member=member, path=extract_directory)
+            files.append(
+                File(
+                    name=member.name, path=os.path.join(extract_directory, member.name)
+                )
+            )
+        tar.close()
+
+        return Directory(name=name, files=files, path=path)
+
+    def __len__(self):
+        return len(self.files)
+
+    def _is_tar(self):
+        _, *extension = self.name.split(".")
```

Comment:
having multiple `.`s outside of extension is valid in file and directory names, for example, we could have `my-model-version-1.5.3-pruned.tar.gz`.  would suggest `extension[-2:]`